### PR TITLE
Pulled in edit-secrets and 040hiera-gpg from nunoc-ops. Added fixes for Ubuntu 20.04.

### PR DESCRIPTION
**Pull request by krihal**
Extended things a bit to allow Ubuntu 20.04 as well as 18.04, can probably be done in a better way so feel free to suggest better solutions. Also pulled in tons of changes from nunoc-ops, I'm not responsible for them!

### CODE REVIEW

File: edit-secrets [hunk 1]
```diff
@@ -66,98 +73,171 @@ if [ "x$1" != "x-l" ]; then
...
+    cmp -s $TMPFILE $TMPFILE2 || status=1
+    if [ $status -eq 0 ]; then
+	(
+	    echo "STATUS=NOT_CHANGED"
+	)  > $LAST_OUTPUT_FILENAME
+	echo ""
+	echo "$0: No changes detected"
+    else
+	# figure out this hosts gpg key id
+	if lsb_release -r | grep -qE '(18|20).04'; then
```
Comment (krihal) [inline edit-secrets:175]: Ubuntu 20 fix.

File: global/pre-tasks.d/040hiera-gpg [hunk 2]
```diff
@@ -6,15 +6,37 @@
...
 export GNUPGHOME
 
+if [ ! -f /usr/bin/eyaml -o ! -d /usr/share/doc/yaml-mode ]; then
+    apt-get update
+    apt-get -y install hiera-eyaml yaml-mode
+fi
+
+if [ ! -f ${EYAMLDIR}/public_certkey.pkcs7.pem -o ! -f ${EYAMLDIR}/private_key.pkcs7.pem ]; then
+    release="x`lsb_release -r | awk '{print $NF}'`"
+    if [ $release = x18.04 ] || [ $release = x20.04 ]; then
```
Comment (krihal) [inline global/pre-tasks.d/040hiera-gpg:20]: Fugly Ubuntu 20 fix.

**Comment by denniswa** [comment]
Basically I just approve anything shiny

**Comment by johanlundberg** [comment]
I would expect fewer changes than this to the nunoc-ops edit secret.

`15:22:45 (kh-fix_ubuntu_20) ~/projects/multiverse$ diff ../nunoc-ops/edit-secrets edit-secrets`

``` diff
79,80c79,81
< 	# remove the STATUS= line
< 	grep -v '^STATUS=EYAML_UPDATED' $TMPFILE > $TMPFILE2
---
> 	# extract the eyaml output
> 	perl -e '$a = 0; while (<>) { $a = 1 if ($_ =~ /^---$/);
>                  print $_ if $a }' < $TMPFILE > $TMPFILE2
82,89c83,85
< 	# check syntax
< 	if [ -x $(dirname $0)/scripts/jsonyaml-no-output.py ]; then
< 	   if ! $(dirname $0)/scripts/jsonyaml-no-output.py yaml $TMPFILE2; then
< 		echo "$0: Error: $TMPFILE2 doesn't look like a YAML file"
< 		exit 1
< 	   fi
< 	else
< 	   echo "$0: Warning: Unable to check syntax of $TMPFILE2"
---
> 	if ! grep "^---$" $TMPFILE2 > /dev/null; then
> 	   echo "$0: Failed extracting yaml output from file $TMPFILE into $TMPFILE2"
> 	   exit 1
179c175
< 	if lsb_release -r | grep -q 18.04; then
---
> 	if lsb_release -r | grep -qE '(18|20).04'; then
```

**Comment by johanlundberg** [comment]
These are the changes if we compare nunoc-ops/global/pre-tasks.d/040hiera-gpg with this one.

`15:34:33 (kh-fix_ubuntu_20) ~/projects/multiverse$ diff ../nunoc-ops/global/pre-tasks.d/040hiera-gpg global/pre-tasks.d/040hiera-gpg` 

``` diff
13,19c13,15
< 
< # There is no hiera-eyaml on Ubuntu < 16.04
< if [ "x`lsb_release -r | awk '{print $NF}'`" != "x12.04" -a "x`lsb_release -r | awk '{print $NF}'`" != "x14.04"  ]; then
<     if [ ! -f /usr/bin/eyaml ]; then
<         apt-get update
<         apt-get -y install hiera-eyaml
<     fi
---
> if [ ! -f /usr/bin/eyaml -o ! -d /usr/share/doc/yaml-mode ]; then
>     apt-get update
>     apt-get -y install hiera-eyaml yaml-mode
22,24c18,21
< if [ -f /usr/bin/eyaml ]; then
<     # Create eyaml keypair if eyaml is installed but there are no keys
<     if [ ! -f ${EYAMLDIR}/public_certkey.pkcs7.pem -o ! -f ${EYAMLDIR}/private_key.pkcs7.pem ]; then
---
> if [ ! -f ${EYAMLDIR}/public_certkey.pkcs7.pem -o ! -f ${EYAMLDIR}/private_key.pkcs7.pem ]; then
>     release="x`lsb_release -r | awk '{print $NF}'`"
>     if [ $release = x18.04 ] || [ $release = x20.04 ]; then
> 	# Create eyaml keypair on Ubuntu 18.04 hosts
27c24
< 	mkdir -p ${EYAMLDIR}
---
> 	mkdir -p /etc/hiera/eyaml
```

**Comment by johanlundberg** [comment]
I guess what I am saying is that the 20.04 fix should be done on nunoc-ops or we should upstream the changes in nunoc-ops to multiverse before applying your fix.

**Comment by eest** [comment]
OK from @krihal to close, nunoc-ops has moved since this was opened, and parts of it was included in https://github.com/SUNET/multiverse/pull/22

**Pull request closed without merging**